### DISCUSSION TITLE
Trace not working.  Need some guidance.

### DISCUSSION THREAD

**Discussion by shatt79** [netbox-community/netbox#12351] [General]
When building a PON circuit where there are patch panels involved, the Z endpoint never shows up when looking at the A interface.  I'm assuming it breaks because I'm connecting the front port of a patch panel to the rear port on the PON splitter.  Is this by design, or am I doing something wrong?  Here's what I'm trying to do:

![pon1](https://user-images.githubusercontent.com/103387334/234629975-cf6531b4-c7da-49c5-8b82-25f8b688d0ed.JPG)

And here's what Netbox shows me on the OLT port.  I want to be able to see splitter (Z) in the Connection field, but it's blank.

![pon2](https://user-images.githubusercontent.com/103387334/234628434-99b2c32b-800a-4abe-905f-292699366fcb.JPG)

When I pull up the trace, it looks complete (attached).  But why can't I get the rear port of the splitter (splitter input) to show up as the Z endpoint when looking at the circuit from the interface view?
[trace.pdf](https://github.com/netbox-community/netbox/files/11334765/trace.pdf)

**Comment by candlerb** [comment]
I'm afraid the way Netbox works today doesn't model PON.

If you associate 1 rear port with 32 front ports, it's actually modelling a 32-strand cable going into the rear port, connecting to 32 different 1-strand cables going out of the front ports - not a splitter where 1 cable splits the same signal 32 ways.  It's intended to model MPO cables between fibre patch panels.

In effect, when you connect an interface to this, only the *first* of the 32 cables is traced.

Enhancements for modelling PON have been requested before, you can probably find them in the issues on github.

**Reply by shatt79** [reply]
I don't really care about tracking ONTs connected to the output ports of the splitters.  I really only want to see what splitter a particular OLT port connects to, and the physical path between.  So, I'm guessing the only way to really make it do what I want is to either modify the splitter device type to include an "interface" and connect my last patch panel to that interface, or set up a new device type to act as a dummy splitter with a single interface only?  Am I correct here?  It's not ideal - as it doesn't accurately reflect an actual PON network - but I guess it'd sorta work...

**Reply by candlerb** [reply]
In the interface table view, "Connection" is a complete, logical interface-to-interface connection only. There's a separate column available "Link Peers" which shows the first hop only (which in this case would be the first patch panel).

In principle, if you were to create an ONT device with an interface, and connect it to the first splitter output, you should see it as a "Connection" to the OLT.

But I think you're right; if you change the splitter to having an 'interface' as an input you'll be able to see it as a connection. If you have multiple splitters in the path you'll only see the first one of course.

I think there might have been proposal at some time for incomplete paths to show the final frontport/rearport in a path as a "connected" peer but I'm not sure.
